### PR TITLE
Kwannoel/create resource refactor

### PR DESCRIPTION
**Pull request by kwannoel**
TODOS:
- [x] Cleanup redux deps
- [x] Allow deletion
- [x] Add component keys
- [x] Scope input to button rather than expose id globally

Out of scope:
- wiring to backend

### CODE REVIEW

File: client/src/components/CreateResourceAddInput.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        dispatch({...action, payload:{ ...action.payload, input: input}})
+    } else {
+        alert("Please Input a valid value! Empty Fields are not permitted")
+    }
+}
+
+// Creates input field and button
+const CreateResourceAddInput = ({dispatch, action}) => {
+    // TODO see if there's a way to scope input
+    const inputId = `create_resource_input${uuidv4()}`
```
Comment (turboMaCk) [inline client/src/components/CreateResourceAddInput.js:20]: I haven't seent this technique of using uuids for `ids` and keys. Doesn't this way of generating them defeat the purpose? I mean I would expect `uuidv4()` call to return different value each time it's called. This means that in every render call different `key` and `ids` will be generated. The purpose of `key` is to have a identifier consistent across multiple render so VDom diffing can correctly map the nodes from old render to the nodes of new render. If this value is different in every render call it essentially tells the mod diffing algorithm to throw everything away and create every dom node from scratch which is both expensive performance-wise and will destroy all local component state in between renders.

Or is this some clever abstraction intended for this purpose which somehow prevents this from happening?

Comment (ivanleomk) [reply client/src/components/CreateResourceAddInput.js:20]: I think the goal of using a UUID was to ensure we had a unique UUID by which to identify individual components. a UUID provides that functionality but we didn't take into account the re-rendering. WHat do you propose in light of this?

File: client/src/components/CreateResourceAddInput.js [hunk 2]
```diff
@@ -17,33 +16,38 @@ function addIfValidInput (input, dispatch, action) {
...
     // TODO see if there's a way to scope input
-    const inputId = `create_resource_input${uuidv4()}`
+    const [userInput, setInput] = React.useState('')
+
+    const handleChange = (e) => {
+        setInput(e.target.value)
+    }
+
+    const handleSubmit = (e) => {
+        e.preventDefault()
```
Comment (kwannoel) [inline client/src/components/CreateResourceAddInput.js:26]: What's this for

File: client/src/components/CreateResourceAddInput.js [hunk 3]
```diff
@@ -17,33 +16,38 @@ function addIfValidInput (input, dispatch, action) {
...
 const CreateResourceAddInput = ({dispatch, action}) => {
     // TODO see if there's a way to scope input
-    const inputId = `create_resource_input${uuidv4()}`
+    const [userInput, setInput] = React.useState('')
+
+    const handleChange = (e) => {
+        setInput(e.target.value)
+    }
+
+    const handleSubmit = (e) => {
```
Comment (kwannoel) [inline client/src/components/CreateResourceAddInput.js:25]: Do we need `e` here, considering that we do not use it in the function body

File: client/src/components/CreateResourceAddInput.js (hunk 2 above)
Comment (ivanleomk) [reply client/src/components/CreateResourceAddInput.js:26]: the e contains information on the button when we click it. By default, form elements will refresh the page when the submit button is clicked. e.preventDefault() stops this. We need this in order to use the form component.

File: client/src/components/CreateResourceAddInput.js (hunk 1 above)
Comment (kwannoel) [reply client/src/components/CreateResourceAddInput.js]: @ivanleomk has updated to use forms

File: client/src/components/CreateResourceAddInput.js (hunk 2 above)
Comment (kwannoel) [reply client/src/components/CreateResourceAddInput.js]: https://www.w3schools.com/jsref/event_preventdefault.asp

https://reactjs.org/docs/forms.html

File: client/src/components/CreateResourceAddInput.js (hunk 3 above)
Comment (kwannoel) [reply client/src/components/CreateResourceAddInput.js]: Resolved in https://github.com/Holmusk/HolmuskDB/pull/68#discussion_r443327071

**Pull request merged**